src/common/modules/AutocorrectHandler.js

```diff
@@ -0,0 +1,203 @@
+import * as emojimart from "/common/lib/emoji-mart-embed/dist/emoji-mart.js";
+
+let autocorrectSymbols = true;
+let autocorrectEmojis = true;
+let autocorrectCharEmojis = false;
+let autocorrectEmojiShortcodes = true;
+let autocomplete = true;
+let quotes = true;
+let fracts = true;
+
+let autocorrections = {};
+
+let longest = 0;
+
+let symbolpatterns = [];
+// Do not autocorrect for these patterns
+let apatterns = [];
+
+let emojiShortcodes = {};
+
+/**
+ * Apply autocorrect settings.
+ *
+ * @returns {void}
+ */
+function settings() {
+	autocorrections = {};
+
+	if (autocorrectSymbols)
+		Object.assign(autocorrections, symbols.symbols);
+	if (autocorrectEmojis) {
+		Object.assign(autocorrections, symbols.emojis);
+		if (autocorrectCharEmojis)
+			Object.assign(autocorrections, symbols.charEmojis);
+	}
+	if (autocorrectEmojiShortcodes) {
+		Object.assign(autocorrections, emojiShortcodes);
+	}
+
+	longest = 0;
+
+	for (const symbol in autocorrections)
+		if (symbol.length > longest)
+			longest = symbol.length;
+	console.log(longest);
+
+	symbolpatterns = [];
+	// Escape special characters
+	const re = /[.*+?^${}()|[\]\\]/g;
+
+	for (const symbol in autocorrections) {
+		symbolpatterns.push(symbol.replace(re, "\\$&"));
+	}
+
+	apatterns = [];
+	for (const x in autocorrections) {
+		let length = 0;
+		let index = x.length;
+
+		for (const y in autocorrections) {
+			if (x !== y) {
+				let aindex = x.indexOf(y);
+				if (aindex >= 0) {
+					if (aindex < index) {
+						index = aindex;
+						length = y.length;
+					} else if (aindex === index && y.length > length)
+						length = y.length;
+				}
+			}
+		}
+
+		if (length > 0) {
+			length = x.length - (index + length);
+			if (length > 1)
+				apatterns.push(x.slice(0, -(length - 1)));
+		}
+	}
+	apatterns = apatterns.filter((item, pos) => apatterns.indexOf(item) === pos);
+	console.log(apatterns);
+
+	apatterns.forEach(function (symbol, index) {
+		apatterns[index] = symbol.replace(re, "\\$&");
+	});
+
+	symbolpatterns = new RegExp('(' + symbolpatterns.join('|') + ')$');
+	apatterns = new RegExp('(' + apatterns.join('|') + ')$');
+}
+
+/**
+ * On error.
+ *
+ * @param {string} error
+ * @returns {void}
+ */
+function onError(error) {
+	console.error(`Error: ${error}`);
+}
+
+/**
+ * Set autocorrect settings.
+ *
+ * @param {Object} pickerResult
+ * @returns {void}
+ */
+function set(pickerResult) {
+	autocorrectEmojis = pickerResult.autocorrectEmojis;
+	autocorrectCharEmojis = pickerResult.autocorrectCharEmojis;
+	autocorrectEmojiShortcodes = pickerResult.autocorrectEmojiShortcodes;
+	autocorrectSymbols = pickerResult.autocorrectSymbols;
+	autocomplete = pickerResult.autocompleteEmojiShortcodes;
+	quotes = pickerResult.autocorrectUnicodeQuotes;
+	fracts = pickerResult.autocorrectUnicodeFracts;
+
+	settings();
+}
+
+/**
+ * Send autocorrect settings to content scripts.
+ *
+ * @param {Object} pickerResult
+ * @returns {void}
+ */
+function apply(pickerResult) {
```

Comment:
"apply What"? Please expand the function name here

Reply:
It applies the settings, by sending them to the content scripts. I will change the function name.